lib/src/domain/user_cases/implementation/get_genres_use_case.dart

```diff
@@ -0,0 +1,17 @@
+import '../../../core/utils/state.dart';
+import '../../../data/datasource/remote/genre_api_provider.dart';
+import '../../../data/datasource/remote/i_api_service.dart';
+import '../../../data/repositories/genre_repository.dart';
+
+class GetGenresUseCase {
+  final ApiService api = GenreApiProvider();
+  late final GenreRepository genreRepo = GenreRepository(
+    api: api,
+  );
+
+  Future<DataState> getGenres() async {
+    return await genreRepo.getData(
+      "",
```

Comment:
This call shouldn't have any parameter

Reply:
If you want to have a default value as an empty string, you can defined on getData method like
`getData([String parameter = ""])`

Reply:
Nice! I didn't know about that way to say that something would be the default value